tests/test_mtdome_csc.py

```diff
@@ -1067,5 +1067,59 @@ async def test_do_setOperationalMode(self) -> None:
                 operational_mode=operational_mode, sub_system_ids=sub_system_ids
             )
 
+    async def test_slow_network(self) -> None:
+        async with self.make_csc(
+            initial_state=salobj.State.STANDBY,
+            config_dir=None,
+            simulation_mode=1,
+            settings_to_apply="",
+        ):
+            await self.set_csc_to_enabled()
+            self.csc.mock_ctrl.slow_network = True
+
+            desired_position = 40
+            desired_velocity = 0.1
+            with salobj.assertRaisesAckError():
+                await self.remote.cmd_moveAz.set_start(
+                    position=desired_position,
+                    velocity=desired_velocity,
+                    timeout=STD_TIMEOUT,
+                )
+            data = await self.assert_next_sample(
+                topic=self.remote.evt_azTarget, position=desired_position
+            )
+            assert desired_velocity == pytest.approx(data.velocity)
+
+    async def test_network_interruption(self) -> None:
+        async with self.make_csc(
+            initial_state=salobj.State.DISABLED,
+            config_dir=None,
+            simulation_mode=1,
+            settings_to_apply="",
+        ):
+            await self.assert_next_summary_state(salobj.State.DISABLED)
+            await self.set_csc_to_enabled()
+            await self.assert_next_summary_state(salobj.State.ENABLED)
+            self.csc.mock_ctrl.network_interruption = True
+            await self.assert_next_summary_state(salobj.State.FAULT)
+
+    async def test_no_connection(self) -> None:
+        # Set simultation_mode deliberately to 0 to make sure that no mock
+        # server is started by the CSC. Also set the initial state to DISABLED
+        # so the RuntimeError, that gets raised when going to ENABLED, can be
+        # caught and processed.
```

Comment:
Is this safe? I'd hate to think that this unit test might succeed in connecting to the real server, if run at the wrong location.

There are safer ways to simulate this, such as adding a flag to the mock controller that makes it refuse all connections (e.g. `enable_server`). It does take a bit of work, especially if you use port 0 for the server and then set it to an available port. But you can report the final port as 0; I think that will reliably make the client fail.

Reply:
That's a very good point that I didn't consider at all! I have introduced a variable in the CSC that can be passed on to the mock controller which, when set to False, will stop the mock controller immediately after starting it. That way, a port gets allocated and the CSC code doesn't break.